src/data/markdown/docs/02 javascript api/07 k6-experimental/01 browser/02 BrowserContext/addCookies.md

```diff
@@ -27,18 +34,47 @@ export const options = {
 
 export default async function () {
   const context = browser.newContext();
+  const page = context.newPage();
 
-  context.addCookies([
-    {
-      name: 'myCookie',
-      value: 'hello world',
-      url: 'https://test.k6.io/',
-    },
-  ]);
+  try {
+    const unixTimeSinceEpoch = Math.round(new Date() / 1000);
+    const day = 60*60*24;
+    const dayAfter = unixTimeSinceEpoch+day;
+    const dayBefore = unixTimeSinceEpoch-day;
 
-  const page = context.newPage();
-  await page.goto('https://test.k6.io/');
-  page.close();
+    context.addCookies([
+      // this cookie expires at the end of the session
+      {
+        name: 'testcookie',
+        value: '1',
+        sameSite: 'Strict',
+        domain: '127.0.0.1',
+        path: '/',
+        httpOnly: true,
+        secure: true,
+      },
+      // this cookie expires in a day
+      {
+        name: 'testcookie2',
+        value: '2',
+        sameSite: 'Lax',
+        domain: '127.0.0.1',
+        path: '/',
+        expires: dayAfter,
+      },
+      // this cookie expires in the past, so it will be removed.
+      {
+        name: 'testcookie3',
+        value: '3',
+        sameSite: 'Lax',
+        domain: '127.0.0.1',
+        path: '/',
+        expires: dayBefore
+      }
+    ]);
```

Comment:
Should we do something else with the page so it's a more realistic example? Such as navigating to some page? In this case seems like we are setting cookies for no actual effect.

Reply:
Sure, what do you suggest?